### DISCUSSION TITLE
OpenAI + Realtime Voice + MAUI Blazor Hybrid - Collaborating?

### DISCUSSION THREAD

**Discussion by cbdg-mitch** [openai/openai-dotnet#305] [General]
Hey gang, I have successfully, or at least like 95% of the way there, developed a MAUI Blazor hybrid implementation of the realtime API using the latest beta open AI nuget. I wanted to see if any one on the OpenAI Nuget team was interested in seeing how I did it, and if down the road there may be a more seamless way to do this. 

I am using Javascript to capture the microphone input and a lot of very custom audio processing. There's slight gaps in the voice response, just a split second here or there, but it works really well overall. I also have an Avatar that is hooked up to it using ThreeJS.

If anyone on this team is interested in looking at how I did it and collaborating on a more MAUI friendly implementation, I'd love to do that with you guys! Not too keen on publishing the work publically as it's a client project, but I'd be happy to add microsoft folks to the repo to take a look at chat about it more.

Thanks! :)

**Comment by edo4444** [comment]
Hey bud! i did the same, a fully Blazor Web Server implementation, works like a charm!
